### ISSUE TITLE
The "Getting Ready" section.

### ISSUE THREAD

**Issue by JeffryGonzalez** [angular-book/angular-book.github.io#3]
I've been sort of ignoring this. Such a can of worms. 

I think it might be best if it was *primarily* links to other content. Like Scott Hanelman's Command Line thing, a "git basics" thing, the "editing code in VSCode" thing. etc. and then maybe some of our content on how this gets used in Angular.

Some things we might have to create content for:

- VS Code
  - Settings (Workspace vs. User, so `.vscode` folder)
  - Recommended settings for doing Angular.
    - Autosave FTW
  - Setting your default terminal for integrated terminal.
  - Code completion, including imports. 
  - A *few* shortcuts: Ctrl+P, Shift+Alt+F, Ctrl+B, stuff like that.
  - Reading error messages and clicking links to the errors.
- Git
  - Using VS Code to make commits, etc. (and those dumb warnings you get the first time)
  - Pushing your code to Github?
  - Branching and Merging (locally, I don't want to get into any workflow stuff, PRs, etc.)
 - Command Line
   - Thinking standardizing on a *sh type shell (GitBash for Windows, WSL, Zsh for Mac, etc.) They are all close enough to the same.
   - Using completions? (does Gitbash have that? Haven't used it in a while)
   - basic navigation (changing directories, the `.` and `..`, 
   - Ctrl+Z
   - Splitting the terminal and multiple terminals in VSCode
 - HTML
   - This stops my brain (and the next two).
   - Maybe *mostly* links to MDN stuff?
   - Maybe some content on the outline model (Accessibility) and angular? Hmmm.
 - CSS
   - Again, mostly MDN links.
   - Maybe introduce using a library like Bootstrap, and a quick intro to Tailwind?
   - Selectors and selector specificity. 
 -  JavaScript
    - "Just enough"
    - Be sure to say that when you are writing TypeScript, you are writing JavaScript, so you'll get more hits if you search for "javascript how to loop through an array" than "typescript how to loop through an array"
    - *Could* do a little Vite project where they do a *bit* of DOM manipulation, adding event handlers, etc.

**Comment by JeffryGonzalez** [comment]
Wondering if the JavaScript CSS thing could just be done in something like Stackblitz?

**Comment by JeffryGonzalez** [comment]
Oh, also - instructions on installing Node, NPM, etc. SEE WHAT I MEAN?

**Comment by lee-cooper** [comment]
One approach would be to teach or hand-hold folks through these topics as they come up in the course. We're not going to teach you these things, but here are resources if you want to know what's going on.

**Comment by JeffryGonzalez** [comment]
> One approach would be to teach or hand-hold folks through these topics as they come up in the course. We're not going to teach you these things, but here are resources if you want to know what's going on. 

Bit confused on your thoughts here - are you missing a "not" before "to teach"? Or are you saying that it would be covered more in a classroom setting?

**Comment by lee-cooper** [comment]
I'm missing context, I believe. When we say "Getting Ready" is this a list of content for folks to read and learn from before class? 

Still getting acclimated to the project, let me know if this makes sense.

**Comment by JeffryGonzalez** [comment]
Yeah, I think I'm losing some context, too. I mean there are some practical things they'd have to do - get node/npm installed, VS Code, etc. Then there is the "skills level-setting" aspect of it. Like how do we say "here's some stuff you need to know before you try to write an Angular application"? Some *basic* command-line stuff, knowing your way around VS Code, the basics of Git/Github, and "just enough" HTML, CSS, JavaScript.

I mean giving links to documentation and stuff is great, but it can also be overwhelming. If you just say "Go read about HTML on MDN" they could take YEARS. haha. 

So, for example, maybe a section on HTML:
- Kinds of elements (inline, block)
- Landmarks (`header`, `article`, `section`, etc.)
- Headings and outline
- The use of `div` and `span`
- Basic forms

I'll see if I can find my notes from the Accessibility course - I had some good stuff in there that maybe we could use.

And CSS:
- Selectors and specificity
- Rules/Properties
- CSS libraries / Tailwind / Ui-Libraries

JavaScript:
- Basic DOM manipulation
- Creating / Removing Elements
- Event Handlers
- Maybe a bit of async - `fetch`, something like that?
-

**Comment by lee-cooper** [comment]
The simplest way is to state the pre-requisites and link to established learning courses outside of the site (e.g. the [web.dev learning site](https://web.dev/learn/) for learning basic html / css / accessibility). 

Because Angular is so opinionated, it's possible to get by without knowing a lot of the basics of Javascript. (speaking from earlier experiences). Folks will see the signposts along the way when they're missing content and (hopefully) seek answers.

Until there's capacity to roll our own courses on these topics, I believe it best to reference other materials and empower the developer to learn and experiment through a series of yo-yo-ing between sources of truth (external learning sites and this book) to learn Angular, and to a greater extent, the nature of web development.
